controllers/postgresql/postgresqluser_controller_test.go

```diff
@@ -0,0 +1,705 @@
+					return errors.New("pgu hasn't been updated by operator")
+				}
+
+				return nil
+			},
+			generalEventuallyTimeout,
+			generalEventuallyInterval,
+		).
+			Should(Succeed())
+
+		// Checks
+		Expect(item.Status.Ready).To(BeTrue())
+		Expect(item.Status.Phase).To(Equal(postgresqlv1alpha1.UserCreatedPhase))
+		Expect(item.Status.Message).To(BeEmpty())
+		Expect(item.Status.RolePrefix).To(Equal("pgu"))
+		Expect(item.Status.PostgresLogin).To(ContainSubstring("pgu-"))
+		Expect(item.Status.PostgresRole).To(ContainSubstring("pgu-"))
+		Expect(item.Status.PostgresGroup).To(Equal(pgdb.Status.Roles.Owner))
+		Expect(item.Status.PostgresDatabaseName).To(Equal(pgdb.Status.Database))
+		Expect(item.Status.LastPasswordChangedTime).ToNot(BeEmpty())
+
+		// Check if user exists
+		postgresRoleExists, postgresRoleErr := isSQLRoleExists(item.Status.PostgresRole)
+		Expect(postgresRoleErr).ToNot(HaveOccurred())
+		Expect(postgresRoleExists).To(BeTrue())
+
+		// Check user is in correct group
+		isMember, err := isSQLUserMemberOf(item.Status.PostgresRole, pgdb.Status.Roles.Owner)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(isMember).To(BeTrue())
+	})
+
+	It("should be ok to change role prefix", func() {
+		// Setup pgec
+		setupPGEC("10s", false)
+
+		// Setup pgdb
+		pgdb := setupPGDB(true)
+
+		// Setup PGU
+		item := setupPGU()
+
+		// Change role prefix
+		pguNewRolePrefix := "pgubis"
+		item.Spec.RolePrefix = pguNewRolePrefix
+
+		Expect(k8sClient.Update(ctx, item)).Should(Succeed())
+
+		updatedItem := &postgresqlv1alpha1.PostgresqlUser{}
+		// Get updated user
+		Eventually(
+			func() error {
+				err := k8sClient.Get(ctx, types.NamespacedName{
+					Name:      pguName,
+					Namespace: pguNamespace,
+				}, updatedItem)
+				// Check error
+				if err != nil {
+					return err
+				}
+
+				// Check if status hasn't been updated
+				if updatedItem.Status.RolePrefix != pguNewRolePrefix {
+					return errors.New("pgu hasn't been updated by operator")
+				}
+
+				return nil
+			},
+			generalEventuallyTimeout,
+			generalEventuallyInterval,
+		).
+			Should(Succeed())
+
+		// Checks
+		Expect(updatedItem.Status.Ready).To(BeTrue())
+		Expect(updatedItem.Status.Phase).To(Equal(postgresqlv1alpha1.UserCreatedPhase))
+		Expect(updatedItem.Status.Message).To(BeEmpty())
+		Expect(updatedItem.Status.RolePrefix).To(Equal(pguNewRolePrefix))
+		Expect(updatedItem.Status.PostgresLogin).To(ContainSubstring(fmt.Sprintf("%s-", pguNewRolePrefix)))
+		Expect(updatedItem.Status.PostgresRole).To(ContainSubstring(fmt.Sprintf("%s-", pguNewRolePrefix)))
+
+		// Check if user exists
+		postgresRoleExists, postgresRoleErr := isSQLRoleExists(updatedItem.Status.PostgresRole)
+		Expect(postgresRoleErr).ToNot(HaveOccurred())
+		Expect(postgresRoleExists).To(BeTrue())
+
+		// Check user is in correct group
+		isMember, err := isSQLUserMemberOf(updatedItem.Status.PostgresRole, pgdb.Status.Roles.Owner)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(isMember).To(BeTrue())
```

Comment:
Can you also check secret values and if there are working ? Also can you test the removal of previous role ?